lib/filewatch/watch.rb

```diff
@@ -18,7 +45,7 @@ def initialize(opts={})
       end
       @watching = []
       @exclude = []
-      @files = Hash.new { |h, k| h[k] = Hash.new }
+      @files = Hash.new { |h, k| h[k] = WatchedFile.new(nil, false) }
```

Comment:
Yeah and with predicates that read better.  Also, it holds the path of the file internally and not just in the `@files.keys`.

It also opens up other possibilities:
- passing it as the yield arg to Tail
- giving it the responsibility of opening and closing its file
- retrieving the Stat.
- making a WatchedFileWindows and WatchedFileNix variants that know what to do in a platform related way.
- maybe removes the necessity to memoize the opened file in the Tail classes